src/Physics/Multinucleon/XSection/NievesSimoVacasMECPXSec2016.cxx

```diff
@@ -339,4 +342,7 @@ void NievesSimoVacasMECPXSec2016::LoadConfig(void)
         dynamic_cast<const XSecIntegratorI *> (
           this->SubAlg("NumericalIntegrationAlg"));
         assert(fXSecIntegrator);
+
+	// Read optional QvalueShifter:
+	fQvalueShifter = dynamic_cast<const QvalueShifter *> ( this->SubAlg("QvalueShifterAlg") );
```

Comment:
This is a bit too simple.
1) initialise the pointer to nullptr
2) check if the Registry contains a shifter algo
3) In case of (2) set the pointer and check that the dynamic cast went through, if not report an error because the desired configuration cannot be satistifed.